File: kommunicate/src/main/java/io/kommunicate/utils/KmInputTextLimitUtil.java
package io.kommunicate.utils;

import io.kommunicate.callbacks.KmCharLimitCallback;

public class KmInputTextLimitUtil {
    private int characterLimit;
    private int notifyAtCharacterCount;

    public KmInputTextLimitUtil(int characterLimit, int charactersRemainingTillLimit) {
        this.characterLimit = characterLimit;
        this.notifyAtCharacterCount = characterLimit - charactersRemainingTillLimit;
    }

    public void checkCharacterLimit(int characterCount, KmCharLimitCallback kmCharLimitCallback) {
        int deltaCharacterCount = characterCount - characterLimit;
        boolean exceeded = characterCount > characterLimit;
        boolean warning = characterCount >= notifyAtCharacterCount;
        if (warning || exceeded) {
            kmCharLimitCallback.onCrossed(exceeded, warning, exceeded ? deltaCharacterCount : -deltaCharacterCount);
        } else {
            kmCharLimitCallback.onNormal();
        }
    }
}

File: kommunicate/src/main/java/io/kommunicate/utils/KmInputTextLimitUtil.kt
package io.kommunicate.utils

import io.kommunicate.callbacks.KmCharLimitCallback

class KmInputTextLimitUtil(private val characterLimit: Int, charactersRemainingTillLimit: Int) {

    private val notifyAtCharacterCount = characterLimit - charactersRemainingTillLimit

    fun checkCharacterLimit(characterCount: Int, kmCharLimitCallback: KmCharLimitCallback) {
        val deltaCharacterCount = characterCount - characterLimit
        val exceeded = characterCount > characterLimit
        val warning = characterCount >= notifyAtCharacterCount
        if (warning || exceeded) {
            kmCharLimitCallback.onCrossed(
                exceeded,
                warning,
                if (exceeded) deltaCharacterCount else -deltaCharacterCount
            )
        } else {
            kmCharLimitCallback.onNormal()
        }
    }
}
